### PR TITLE
feat: add support for synchronous environments

### PR DESCRIPTION
**Pull request by ahal**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,6 +46,16 @@ async with TokenClient(token) as session:
     await session.get("/octocat")
 ```
 
+If calling synchronously, simply remove the `async` / `await` from the
+examples:
```
Comment (jcristau) [inline README.md:50]: Is this a common pattern, vs just having separate sync and async APIs?  It feels a bit too clever somehow...

Comment (ahal) [reply README.md:50]: I've seen it.. but really there are two APIs and I guess you could instantiate `SyncClient` or `AsyncClient` directly if you wanted. The `TokenClient` and `AppClient` functions are just convenience wrappers around those.

Is there a downside to it? The code to detect it is pretty simple.

Comment (jcristau) [reply README.md:50]: It's pretty simple but I'm not sure it's correct?  When running `client = TokenClient(token)` you don't know whether that'll be followed by `async with client as session: ...` e.g., you'd have to wait for either `__enter__` or `__aenter__` gets called to decide?

Comment (ahal) [reply README.md:50]: If you're in an asynchronous environment and try to use it synchronously (or vice versa) it'll fail. That seems expected to me?

Comment (ahal) [reply README.md:50]: To try and re-word my previous comment..

If you use the `TokenClient` or `AppClient` shortcuts inside a running event loop, you *have* to use it asynchronously. E.g:

```python
# inside event loop
client = TokenClient(token)

async with client as session:  # ok
    ...
    
with client as session:  # Exception is raised
    ...
```

So there's no decision that `simple-github` needs to make around `__enter__` or `__aenter__`, that's delegated to the user.

This does require the user to be aware of whether they are in an asynchronous context or not, but that doesn't feel like a heavy burden to ask. It's probably even a feature that we enforce they use an AsyncClient in this case. If for some strange reason they want to use the client synchronously anyway, they can always do:

```python
# inside event loop
auth = TokenAuth(token)
client = ClientSync(auth=auth)
with client as session:  # ok
    ...
```

But I don't know why that would ever be desirable.

**Pull request merged**